File: Frontend/src/app/business-logic/SSLFactors.ts
import { Vulnerability } from "./Vulnerability";

export class SSLFactors {
    public grade:String | undefined;
    public isSelfSigned:boolean|undefined;
    public isExpired:boolean|undefined;
    public isWeakCipher:boolean|undefined;
    public vulnerabilities:Array<Vulnerability> | undefined;
    public  issuedTo:String | undefined;
    public  issuedBy:String | undefined;
    public  cipher:String | undefined;
    public  issuedCountry:String | undefined;
    public  validFrom:String | undefined;
    public  validTill:String | undefined;

    public SSLFactors() {
        this.grade = "";
        this.isSelfSigned = false;
        this.isExpired = false;
        this.isWeakCipher = false;
        this.vulnerabilities = new Array<Vulnerability>();
        this.issuedBy = "";
        this.issuedTo = "";
        this.cipher = "";
        this.issuedCountry = "";
        this.validFrom = "";
        this.validTill = "";
    }


}

File: Backend/src/main/java/com/example/riskfactors/model/SSLFactors.java
package com.example.riskfactors.model;

import lombok.*;
import javax.persistence.*;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.SequenceGenerator;
import java.util.ArrayList;
import java.util.List;
@Setter
@Getter
@Entity
@Data
@AllArgsConstructor
@Builder
public class SSLFactors {


    @Id
    @SequenceGenerator(
            name = "SSLFactorsSequence",
            sequenceName = "SSLFactorsSequence",
            allocationSize = 1
    )
    @GeneratedValue(
            strategy = GenerationType.SEQUENCE,
            generator = "SSLFactorsSequence"
    )
    private Long SSLFactorsId;
    private String grade;
    private Boolean isSelfSigned;
    private Boolean isExpired;
    private Boolean isWeakCipher;

    @OneToMany(
            cascade = CascadeType.ALL
    )
    @JoinColumn(
            name = "vulnerabilityFK"
    )
    private List<Vulnerability> vulnerabilities;
    private String issuedTo;
    private String issuedBy;
    private String cipher;
    private String issuedCountry;
    private String validFrom;
    private String validTill;

    public SSLFactors() {
        this.grade = "";
        this.isSelfSigned = false;
        this.isExpired = false;
        this.isWeakCipher = false;
        this.vulnerabilities = new ArrayList<>();
        this.issuedBy = "";
        this.issuedTo = "";
        this.cipher = "";
        this.issuedCountry = "";
        this.validFrom = "";
        this.validTill = "";
    }


}
